src/test/java/uk/gov/justice/digital/zone/StructuredZoneTest.java

```diff
@@ -0,0 +1,189 @@
+        doReturn(Optional.of("s3://loadjob/structured")).when(jobParams).getStructuredS3Path();
+        StructuredZone structuredZoneTest = spy(new StructuredZone(jobParams, storage1));
+        doReturn(mockedDataSet).when(structuredZoneTest).validateJsonData(any(), any(), any(), any());
+        doNothing().when(structuredZoneTest).handleInValidRecords(any(), any(), any(), any(), any());
+        doReturn(mockedDataSet).when(structuredZoneTest).handleValidRecords(any(), any(), any());
+        when(mockedDataSet.count()).thenReturn(10L);
+        Dataset<Row> actual_result = structuredZoneTest.process(spark, mockedDataSet, table);
+        assertNotNull(actual_result);
+    }
+
+    @Test
+    @ExtendWith(MockitoExtension.class)
+    public void shouldHandleInValidRecords() {
+        // Define a schema for the row
+        StructType schema = new StructType()
+                .add("source", StringType, false)
+                .add("table", StringType, false)
+                .add("operation", StringType, false);
+        // Create a Row object with key-value pairs
+        Row table = new GenericRowWithSchema(Arrays.asList("oms_owner", "agency_internal_locations", "load").toArray(),
+                schema);
+        JobParameters jobParams = mock(JobParameters.class);
+        DataStorageService storage1 = mock(DataStorageService.class);
+        sourceRefStatic.when(() -> SourceReferenceService
+                        .getSourceReference(table.getAs("source"), table.getAs("table")))
+                .thenReturn(optionalRef);
+        doReturn(false).when(optionalRef).isPresent();
+
+        doReturn(Optional.of("s3://loadjob/violations")).when(jobParams).getViolationsS3Path();
+        doReturn(Optional.of("s3://loadjob/structured")).when(jobParams).getStructuredS3Path();
+        StructuredZone structuredZoneTest = spy(new StructuredZone(jobParams, storage1));
+        doReturn(mockedDataSet).when(structuredZoneTest).handleNoSchemaFound(any(), any(), any(), any());
+        when(mockedDataSet.count()).thenReturn(10L);
+        Dataset<Row> actual_result = structuredZoneTest.process(spark, mockedDataSet, table);
+        assertNotNull(actual_result);
+    }
+
+    @Test
+    @ExtendWith(MockitoExtension.class)
+    public void shouldHandleSchemaFound() {
+        DataStorageService storage1 = mock(DataStorageService.class);
+        JobParameters jobParams = mock(JobParameters.class);
+        doReturn(Optional.of("s3://loadjob/violations")).when(jobParams).getViolationsS3Path();
+        doReturn(Optional.of("s3://loadjob/structured")).when(jobParams).getStructuredS3Path();
+        StructuredZone structuredZoneTest = spy(new StructuredZone(jobParams, storage1));
+        Optional<SourceReference> optionalRef = SourceReferenceService.getSourceReference("oms_owner",
+                "offenders");
+        SourceReference ref = optionalRef.orElse(null);
+
+        if (ref != null) {
```

Comment:
This if block hides the unexpected case where ref is null. Remove the if and replace it with an `assertNotNull`